python/cudf/cudf/core/udf/__init__.py

```diff
@@ -25,35 +25,35 @@
 _STRING_UDFS_ENABLED = False
 try:
     import strings_udf
+    from strings_udf import ptxpath
 
-    if strings_udf.ENABLED:
-        from . import strings_typing  # isort: skip
-        from . import strings_lowering  # isort: skip
-        from strings_udf import ptxpath
-        from strings_udf._lib.cudf_jit_udf import to_string_view_array
-        from strings_udf._typing import str_view_arg_handler, string_view
-
-        # add an overload of MaskedType.__init__(string_view, bool)
-        cuda_lower(api.Masked, strings_typing.string_view, types.boolean)(
-            masked_lowering.masked_constructor
-        )
-
-        # add an overload of pack_return(string_view)
-        cuda_lower(api.pack_return, strings_typing.string_view)(
-            masked_lowering.pack_return_scalar_impl
-        )
-
-        _supported_masked_types |= {strings_typing.string_view}
-        utils.launch_arg_getters[dtype("O")] = to_string_view_array
-        utils.masked_array_types[dtype("O")] = string_view
-        utils.JIT_SUPPORTED_TYPES |= STRING_TYPES
+    if ptxpath:
         utils.ptx_files.append(ptxpath)
```

Comment:
My understanding of this logic was that if rapids was not initialized, one wouldn't expect to be able to use any of its runtime functionality in the first place - but one should still continue initializing the rest of the library so its functions and modules are importable. I took inspiration from the following. https://github.com/rapidsai/cudf/blob/7d173c9d144a64c5e1a0467d2a5eb4181854f25e/python/cudf/cudf/utils/gpu_utils.py#L10-L14

Reply:
The rationale in that function is to avoid creating a cuda context at import time (which causes problems with the UCX networking stack), so in `validate_setup` if we said `RAPIDS_NO_INITIALIZE` then we just blindly continue (everything is usable but we don't validate the GPU configuration). 

In this case we're doing some amount of auto-discovery of functionality at import time (which again creates a cuda context which we'd like to try and avoid). If we can't discover the functionality I suppose that we should turn it off. Does that happen here?

A more involved thing (which we should not do in this PR) would be to move the auto-discovery to a function that is called once the first time strings_udf functionality is used (e.g. by having a zero-argument function decorated by `@functools.cache`). In that situation, if `RAPIDS_NO_INITIALIZE` is in the environment we can still check if we _already_ have a context (in which case we're allowed to query device attributes for the associated device).

Reply:
how would you feel about `strings_udf` propagating an `ImportError` if `ptxpath` cannot be resolved? That'd cause us to fall into the `pass` case down below, which would give the user back a version of cudf with `strings_udf` disabled.